cpu/esp8266/Makefile.include

```diff
@@ -131,7 +131,6 @@ endif
 
 LINKFLAGS += -T$(RIOTCPU)/$(CPU)/ld/eagle.rom.addr.v6.ld
 LINKFLAGS += -nostdlib -lgcc -u ets_run -Wl,-gc-sections # -Wl,--print-gc-sections
-LINKFLAGS += -Wl,--warn-unresolved-symbols
```

Comment:
@cladmi I added this link flag by intention. I don't remember exactly, but the reason was that the compilation of a lot of test applications or examples failed.

Reply:
From the errors I see I understand where it comes from :)
The goal of this tracking pr is to try finding them and solving them if they are bugs.

The drawback I have with having this option, is I used "stdio_read" in a function and was able to generate a firmware that passed the compilation (with a warning but when there is one, why not 10) but failed at runtime. I was indeed missing a dependency but the build did not fail because of this. So we can link and generate invalid firmwares.

I however do not currently understand why the other platforms manage to link without providing errors for these modules. Maybe a difference in a flag but I did not investigate for the moment.

Reply:
@cladmin
The cause of undefined references to `pthread_setcancelstate` is a quite weird. It happens for all applications where the `pthread` module is used. For ESP32, the `pthread` module is activated automatically whenever the `core_thread_flags` module is used.

What I don't understand is that these undefined references happen although the `pthread` module provides a defined symbol `pthread_setcancelstate` even though the function is only a dummy function which does nothing.